Remove local `track` tests in favor of WPT

Ahmad-S792 opened the pull request:
#### 1fcbc136b4143a32dca6429fa0679c24cd41e763
<pre>
Remove local `track` tests in favor of WPT

<a href="https://bugs.webkit.org/show_bug.cgi?id=269655">https://bugs.webkit.org/show_bug.cgi?id=269655</a>

Reviewed by Eric Carlson.

This is preliminary effort to de-duplicate bunch of local tests in
`media` specifically `track` in favor of WPT tests:

&gt; Renamed `vtt-region-parser.html ` to WPT `header-regions.html`
* LayoutTests/media/track/regions-webvtt/vtt-region-parser.html: Deleted
* LayoutTests/media/track/regions-webvtt/vtt-region-parser-expected.txt: Ditto
Reference: <a href="https://chromium.googlesource.com/chromium/src.git/+/ad11f82eeaba1fb9ad63c45809825d840e7ce44c">https://chromium.googlesource.com/chromium/src.git/+/ad11f82eeaba1fb9ad63c45809825d840e7ce44c</a>

&gt; Removed local copies `track-cues-cuechange.html` and
`track-cues-enter-exit.html` in favor of WPT with same name
&gt; Deleted `resource` file `cues-chrono-order.vtt` as well
* LayoutTests/media/track/captions-webvtt/cues-chrono-order.vtt: Deleted
* LayoutTests/media/track/track-cues-cuechange.html: Deleted
* LayoutTests/media/track/track-cues-cuechange-expected.txt: Ditto
* LayoutTests/media/track/track-cues-enter-exit.html: Ditto
* LayoutTests/media/track/track-cues-enter-exit-expected.txt: Ditto
Reference: <a href="https://chromium.googlesource.com/chromium/src.git/+/abe01ca5f57b5cf02dcc0d65a3c022d599eb56c3">https://chromium.googlesource.com/chromium/src.git/+/abe01ca5f57b5cf02dcc0d65a3c022d599eb56c3</a>

&gt; Deleted `track-node-add-remove.html` in favor of WPT Test with same name
* LayoutTests/media/track/track-node-add-remove.html: Deleted
* LayoutTests/media/track/track-node-add-remove-expected.txt: Ditto
Reference: <a href="https://chromium.googlesource.com/chromium/src.git/+/c2009a32aabb712fbba4955ac0e86986e56e9ad4">https://chromium.googlesource.com/chromium/src.git/+/c2009a32aabb712fbba4955ac0e86986e56e9ad4</a>

&gt; Remove local copies of below tests in favor of WPT:
* LayoutTests/media/track/track-remove-track.html: With same name
* LayoutTests/media/track/track-remove-track-expected.txt:
* LayoutTests/media/track/track-remove-active-cue-crash.html: With &apos;track-remove-active-cue.html&apos; name
* LayoutTests/media/track/track-remove-active-cue-crash-expected.txt:
* LayoutTests/media/track/track-node-add-remove.html: With same name
* LayoutTests/media/track/track-node-add-remove-expected.txt:
* LayoutTests/media/track/track-cues-enter-exit.html: With same name
* LayoutTests/media/track/track-cues-enter-exit-expected.txt:
* LayoutTests/media/track/track-active-cues.html: With same name
* LayoutTests/media/track/track-active-cues-expected.txt

&gt; Removed `crossOrigin.html` in favor of WPT with same name
* LayoutTests/media/track/opera/interfaces/HTMLElement/HTMLMediaElement/crossOrigin.html: Deleted
Reference: <a href="https://chromium.googlesource.com/chromium/src.git/+/8dea498d5fcc87730ac1f9e294d08a053184e68c">https://chromium.googlesource.com/chromium/src.git/+/8dea498d5fcc87730ac1f9e294d08a053184e68c</a>

&gt; Removed `track-constants.html` now in `readyState.html` on WPT
* LayoutTests/media/track/track-constants.html: Deleted
* LayoutTests/media/track/track-constants-expected.txt: Deleted
Reference: <a href="https://chromium.googlesource.com/chromium/src.git/+/b6f9344a2370bc3970643a2a532655fffe5568e9">https://chromium.googlesource.com/chromium/src.git/+/b6f9344a2370bc3970643a2a532655fffe5568e9</a>

&gt; Remove local `track-webvtt-tc[001-032]-*.html` in favor of WPT:
* WPT Directory - &apos;html/semantics/embedded-content/media-elements/track/track-element&apos;
and starting with &apos;track-webvtt-&apos; (e.g. track-webvtt-tc001-utf8.html
-&gt; track-webvtt-utf8.html)
Reference: <a href="https://chromium.googlesource.com/chromium/src.git/+/253def87b2e0d66fe0af3f8bce11409469668858">https://chromium.googlesource.com/chromium/src.git/+/253def87b2e0d66fe0af3f8bce11409469668858</a>
<a href="https://chromium.googlesource.com/chromium/src.git/+/21025b8461ad6d5d90bdfa296171bddd755afd5f">https://chromium.googlesource.com/chromium/src.git/+/21025b8461ad6d5d90bdfa296171bddd755afd5f</a>
<a href="https://chromium.googlesource.com/chromium/src.git/+/a14d026f402dfc66216603926eff6593be4e012c">https://chromium.googlesource.com/chromium/src.git/+/a14d026f402dfc66216603926eff6593be4e012c</a>
<a href="https://chromium.googlesource.com/chromium/src.git/+/bc34071d92d7dbcd5425d7b8991b6f0399b351b1">https://chromium.googlesource.com/chromium/src.git/+/bc34071d92d7dbcd5425d7b8991b6f0399b351b1</a>
<a href="https://chromium.googlesource.com/chromium/src.git/+/d33b073d775c0d21cfb94f54da743c1015acf93d">https://chromium.googlesource.com/chromium/src.git/+/d33b073d775c0d21cfb94f54da743c1015acf93d</a>
<a href="https://chromium.googlesource.com/chromium/src.git/+/20fe39102f1ee7c1548ecf43937a0cf07a4666b0">https://chromium.googlesource.com/chromium/src.git/+/20fe39102f1ee7c1548ecf43937a0cf07a4666b0</a>
* Remove local resource files as well from: `LayoutTests/media/track/captions-webvtt/`
(for above tests)

&gt; Update &apos;TestExpectation&apos; files:
* LayoutTests/platform/glib/TestExpectations: Updated
* LayoutTests/platform/mac-wk1/TestExpectations: Ditto
* LayoutTests/platform/mac/TestExpectations: Ditto

Canonical link: <a href="https://commits.webkit.org/274948@main">https://commits.webkit.org/274948@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/646c925987461da0c6dfedafeec3579872341f8a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40457 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19469 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42835 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43007 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/36546 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22424 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16798 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33567 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41031 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16422 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34880 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14151 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14217 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35852 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44283 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36683 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36186 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39920 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15272 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12522 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38220 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16891 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9072 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16941 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16535 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson reviewed (approved):
Very nice, thank you!

Ahmad-S792 committed:
Remove local `track` tests in favor of WPT

https://bugs.webkit.org/show_bug.cgi?id=269655

Reviewed by Eric Carlson.

This is preliminary effort to de-duplicate bunch of local tests in
`media` specifically `track` in favor of WPT tests:

> Renamed `vtt-region-parser.html ` to WPT `header-regions.html`
* LayoutTests/media/track/regions-webvtt/vtt-region-parser.html: Deleted
* LayoutTests/media/track/regions-webvtt/vtt-region-parser-expected.txt: Ditto
Reference: https://chromium.googlesource.com/chromium/src.git/+/ad11f82eeaba1fb9ad63c45809825d840e7ce44c

> Removed local copies `track-cues-cuechange.html` and
`track-cues-enter-exit.html` in favor of WPT with same name
> Deleted `resource` file `cues-chrono-order.vtt` as well
* LayoutTests/media/track/captions-webvtt/cues-chrono-order.vtt: Deleted
* LayoutTests/media/track/track-cues-cuechange.html: Deleted
* LayoutTests/media/track/track-cues-cuechange-expected.txt: Ditto
* LayoutTests/media/track/track-cues-enter-exit.html: Ditto
* LayoutTests/media/track/track-cues-enter-exit-expected.txt: Ditto
Reference: https://chromium.googlesource.com/chromium/src.git/+/abe01ca5f57b5cf02dcc0d65a3c022d599eb56c3

> Deleted `track-node-add-remove.html` in favor of WPT Test with same name
* LayoutTests/media/track/track-node-add-remove.html: Deleted
* LayoutTests/media/track/track-node-add-remove-expected.txt: Ditto
Reference: https://chromium.googlesource.com/chromium/src.git/+/c2009a32aabb712fbba4955ac0e86986e56e9ad4

> Remove local copies of below tests in favor of WPT:
* LayoutTests/media/track/track-remove-track.html: With same name
* LayoutTests/media/track/track-remove-track-expected.txt:
* LayoutTests/media/track/track-remove-active-cue-crash.html: With 'track-remove-active-cue.html' name
* LayoutTests/media/track/track-remove-active-cue-crash-expected.txt:
* LayoutTests/media/track/track-node-add-remove.html: With same name
* LayoutTests/media/track/track-node-add-remove-expected.txt:
* LayoutTests/media/track/track-cues-enter-exit.html: With same name
* LayoutTests/media/track/track-cues-enter-exit-expected.txt:
* LayoutTests/media/track/track-active-cues.html: With same name
* LayoutTests/media/track/track-active-cues-expected.txt

> Removed `crossOrigin.html` in favor of WPT with same name
* LayoutTests/media/track/opera/interfaces/HTMLElement/HTMLMediaElement/crossOrigin.html: Deleted
Reference: https://chromium.googlesource.com/chromium/src.git/+/8dea498d5fcc87730ac1f9e294d08a053184e68c

> Removed `track-constants.html` now in `readyState.html` on WPT
* LayoutTests/media/track/track-constants.html: Deleted
* LayoutTests/media/track/track-constants-expected.txt: Deleted
Reference: https://chromium.googlesource.com/chromium/src.git/+/b6f9344a2370bc3970643a2a532655fffe5568e9

> Remove local `track-webvtt-tc[001-032]-*.html` in favor of WPT:
* WPT Directory - 'html/semantics/embedded-content/media-elements/track/track-element'
and starting with 'track-webvtt-' (e.g. track-webvtt-tc001-utf8.html
-> track-webvtt-utf8.html)
Reference: https://chromium.googlesource.com/chromium/src.git/+/253def87b2e0d66fe0af3f8bce11409469668858
https://chromium.googlesource.com/chromium/src.git/+/21025b8461ad6d5d90bdfa296171bddd755afd5f
https://chromium.googlesource.com/chromium/src.git/+/a14d026f402dfc66216603926eff6593be4e012c
https://chromium.googlesource.com/chromium/src.git/+/bc34071d92d7dbcd5425d7b8991b6f0399b351b1
https://chromium.googlesource.com/chromium/src.git/+/d33b073d775c0d21cfb94f54da743c1015acf93d
https://chromium.googlesource.com/chromium/src.git/+/20fe39102f1ee7c1548ecf43937a0cf07a4666b0
* Remove local resource files as well from: `LayoutTests/media/track/captions-webvtt/`
(for above tests)

> Update 'TestExpectation' files:
* LayoutTests/platform/glib/TestExpectations: Updated
* LayoutTests/platform/mac-wk1/TestExpectations: Ditto
* LayoutTests/platform/mac/TestExpectations: Ditto

Canonical link: https://commits.webkit.org/274948@main
